homeassistant/components/caldav/todo.py

```diff
@@ -165,6 +159,15 @@ async def async_update_todo_item(self, item: TodoItem) -> None:
             vtodo["DESCRIPTION"] = description
         else:
             vtodo.pop("DESCRIPTION", None)
+        if item.status == TodoItemStatus.COMPLETED:
```

Comment:
If the item is already completed and an update arrives to, say, edit the summar, this appears like it will mark the item completed again and reset the completion timestamp